src/net/client/load_balancer.rs

```diff
@@ -0,0 +1,1140 @@
+                                            if res.0 == ind {
+                                                // The current upstream finished,
+                                                // try the next one, if any.
+                                                self.state =
+                                                if ind+1 < self.conn_rt.len() {
+                                                    QueryState::Probe(ind+1)
+                                                }
+                                                else
+                                                {
+                                                    QueryState::Wait
+                                                };
+                                                // Break out of receive loop
+                                                break;
+                                            }
+                                            // Just continue receiving
+                                            continue;
+                                        }
+                                        // Return error to the user.
+                                    }
+                                    Ok(ref msg) => {
+                                        if skip(msg, &self.config) {
+                                            if self.deferred_reply.is_none() {
+                                                self.deferred_reply = Some(msg.clone());
+                                            }
+                                            if res.0 == ind {
+                                                // The current upstream finished,
+                                                // try the next one, if any.
+                                                self.state =
+                                                    if ind+1 < self.conn_rt.len() {
+                                                        QueryState::Probe(ind+1)
+                                                    }
+                                                    else
+                                                    {
+                                                        QueryState::Wait
+                                                    };
+                                                // Break out of receive loop
+                                                break;
+                                            }
+                                            // Just continue receiving
+                                            continue;
+                                        }
+                                        // Now we have a reply that can be
+                                        // returned to the user.
+                                    }
+                                }
+                                self.result = Some(res.1);
+                                self.res_index= res.0;
```

Comment:
Missing whitespace - run `cargo fmt`?

Reply:
Funny, this must be a bug in fmt.

Reply:
Ah, inside select!. I added the required space.